### PR TITLE
Hookshot Anywhere

### PR DESCRIPTION
**Pull request by cplaster**
Allows most surfaces to be hookshot-able.

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2053472381.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2053490656.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2053527439.zip)

### CODE REVIEW

**Comment by inspectredc** [comment]
Hi! It looks as though you may have committed some merge conflict files in case you hadn't noticed!

**Comment by cplaster** [comment]
Oops! I hadn't noticed. I think this should be good now!

**Comment by mckinlee** [comment]
Just a FYI cause it could just be me and my environment, but this doesn't appear to work.

File: mm/2s2h/Enhancements/GameInteractor/GameInteractor.h [hunk 1]
```diff
@@ -55,6 +55,7 @@ typedef enum {
     GI_VB_FLIP_HOP_VARIABLE,
     GI_VB_DISABLE_LETTERBOX,
     GI_VB_CLOCK_TOWER_OPENING_CONSIDER_THIS_FIRST_CYCLE,
+    GI_VB_ENABLE_HOOKSHOT_ANYWHERE,
```
Comment (garrettjoecox) [inline mm/2s2h/Enhancements/GameInteractor/GameInteractor.h]:
```suggestion
    GI_VB_BE_HOOKSHOT_SURFACE,
```

File: mm/src/code/z_bgcheck.c [hunk 2]
```diff
@@ -4266,6 +4267,9 @@ u32 SurfaceType_GetEcho(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId)
 }
 
 u32 SurfaceType_IsHookshotSurface(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+    if (GameInteractor_Should(GI_VB_ENABLE_HOOKSHOT_ANYWHERE, false, NULL))
+        return 1;
+
     return SurfaceType_GetData(colCtx, poly, bgId, 1) >> 17 & 1;
```
Comment (garrettjoecox) [inline mm/src/code/z_bgcheck.c]: This way, someone could use the same hook to make _no_ surfaces hookshotable, or maybe _metal_ surfaces hookshotable etc
```suggestion
    return GameInteractor_Should(GI_VB_BE_HOOKSHOT_SURFACE, SurfaceType_GetData(colCtx, poly, bgId, 1) >> 17 & 1, poly, bgId);
```

**Review by garrettjoecox — COMMENTED** [review]
Left a few suggestions, also needs updating to use the new menu & header pattern

File: mm/src/code/z_bgcheck.c (hunk 2 above)
Comment (cplaster) [reply mm/src/code/z_bgcheck.c]: Great idea! Updated with your suggestions.

**Comment by garrettjoecox** [comment]
Did you have trouble resolving the conflicts here? I can help out if not to lose this contribution!

**Comment by cplaster** [comment]
My branch of develop was ~3 months behind, tried to rebase. Fixed the merge conflicts, but it won't compile because it complains about the libultraship submodule now, and I gave up trying to fix it. I have another look after the weekend.

**Comment by cplaster** [comment]
Reopened PR, I think I've got this up to snuff now.

**Review by Archez — APPROVED** [review]
Hope you don't mind, but I pushed a small commit to help get this across the finish line.
Namely, ran the code formatter and aligned to use the new menu/headers.

**Pull request merged**